<!doctype html>
<!-- The Time Machine GitHub pages theme was designed and developed by Jon Rohan, on Feb 7, 2012. -->
<!-- Follow him for fun. http://twitter.com/jonrohan. Tail his code on https://github.com/jonrohan -->
<html>
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

  <link rel="stylesheet" href="stylesheets/stylesheet.css" media="screen">
  <link rel="stylesheet" href="stylesheets/github-dark.css">
  <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
  <script type="text/javascript" src="javascripts/script.js"></script>

  <title>Juzhe1120.GitHub.com</title>
  <meta name="description" content="This MATLAB software package is used to predict pupylation sites in prokaryotic proteins.">

  <meta name="viewport" content="width=device-width,initial-scale=1">

</head>

<body>

  <div class="wrapper">
    <header>
      <h1 class="title">Juzhe1120.GitHub.com</h1>
    </header>
    <div id="container">
      <p class="tagline">Predicting pupylation sites in prokaryotic proteins using semi-supervised self-training support vector machine algorithm</p>
      <div id="main" role="main">
        <div class="download-bar">
        <span class="blc"></span><span class="trc"></span>
        </div>
        <article class="markdown-body">
          <h3>
<a id="welcome-to-github-pages" class="anchor" href="#welcome-to-github-pages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Introduction</h3>

<p>As one important post-translational modification of prokaryotic proteins, pupylation plays a key role in regulating various biological processes. The accurate identification of pupylation sites is crucial for understanding the underlying mechanisms of pupylation. Although several computational methods have been developed for the identification of pupylation sites, the prediction accuracy of them is still unsatisfactory. Here, a novel bioinformatics tool named IMP-PUP is proposed to improve the prediction of pupylation sites. IMP-PUP is constructed on the composition of k-spaced amino acid pairs and trained with a modified semi-supervised self-training support vector machine (SVM) algorithm. The proposed algorithm iteratively trains a series of support vector machine classifiers on both annotated and non-annotated pupylated proteins. Computational results show that IMP-PUP achieves the area under receiver operating characteristic curve of 0.91, 0.73 and 0.75 on our training set, Tung's testing set and our testing set, respectively, which are better than those of Different Error Cost SVM algorithm and original self-training SVM algorithm. Independent tests also show that IMP-PUP significantly outperforms other three existing pupylation sites predictors GPS-PUP, iPUP and pbPUP. Therefore, IMP-PUP can be a useful tool for accurate prediction of pupylation sites.</p>

<h3>
<a id="designer-templates" class="anchor" href="#designer-templates" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>IMP-PUP software MATLAB code</h3>

<p>The whole software MATLAB code of IMP-PUP is availabe by clicking <a href="http://github.com/juzhe1120/Matlab_Software/blob/master/IMP-PUP_Matlab_Software.zip?raw=true"><b>here</b></a>.</p>

<p>Usage of this MATLAB software package:
<ol>
<li> Prepare your sequence(s) file in fasta format and name it "Xinput.fasta"</li>

<li> Run the program "IMP_PUP.m"</li>

<li>Get the result file "Pre_result.xls"</li>
</ol>
</p>
<p>Please note that this software package only supports the 32-bit version of Windows operating system.</p>
<h3>
<a id="creating-pages-manually" class="anchor" href="#creating-pages-manually" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Dataset</h3>

<p>Our training set, our independent testing set, Tung's independent testing set and unlabeled training set are availabe by clicking   <a href="https://github.com/juzhe1120/juzhe1120.github.com/blob/master/Dataset.xls?raw=true"><b>here</b>.</p>
<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Reference</h3>

<p>Zhe Ju, Hong Gu, Predicting pupylation sites in prokaryotic proteins using semi-supervised self-training support vector machine algorithm, *Analytical Biochemistry* (2016), doi: 10.1016/j.ab.2016.05.005.</p>

<h3>
<a id="support-or-contact" class="anchor" href="#support-or-contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a></h3>
<p>If you have any questions, please contact <a href="mailto:juzhe1120@hotmail.com">Zhe Ju</a></p>

        </article>
      </div>
    </div>
    <footer>
      <div class="owner">

      <p><a href="https://github.com/juzhe1120" class="avatar"><img src="https://avatars2.githubusercontent.com/u/12395206?v=3&amp;s=60" width="48" height="48"></a>View <a href="https://github.com/juzhe1120">juzhe1120</a> on <a href="https://www.github.com">GitHub</a></p>

      </div>
      <div class="creds">
        <small>This page generated using <a href="https://pages.github.com/">GitHub Pages</a><br>theme by <a href="https://twitter.com/jonrohan/">Jon Rohan</a></small>
      </div>
    </footer>
  </div>
  <div class="current-section">
    <a href="#top">Scroll to top</a>
    <a href="https://github.com/juzhe1120/juzhe1120.github.com/tarball/master" class="tar">tar</a><a href="https://github.com/juzhe1120/juzhe1120.github.com/zipball/master" class="zip">zip</a><a href="" class="code">source code</a>
    <p class="name"></p>
  </div>

  
</body>
</html>
